MetaMask/metamask-extension
# ci(chore): Run snaps tests in main unit test job

## **Description**
This pull request starts the process of moving E2E tests to be ran by build type, focusing on snaps and accounts first. Snaps and accounts tests were being ran on the flask build only, but the name of the e2e test script was flask-snaps which was running all of the flask test suite, plus accounts, plus snaps. This was achieved by adding a --build-type flag to the command in the circle ci script which was `yarn test:e2e:chrome:snaps` the snaps part added a --snaps flag to the e2e test command. This PR:

1. removes the `--snaps` flag from the E2E test command, instead looking only at the `--build-type` flag to determine if flask versus main tests should be run. 
2. Runs most snaps and accounts tests against both main and flask as flask is still a developing ground for new features. I might recommend we improve this logic in the future by using a combination of extension '.flask.spec.js' and '@flask' so that existing features that are included in main can be extended in flask with new tests built for them instead of the current logic. The current logic just filters out specific tests (pre-existing just moved to a new location in the code).
3. Increases concurrency of e2e test runners. 
4. the `test-e2e-chrome-snap-flask` and `test-e2e-firefox-snaps-flask` has been renamed to drop the `-snaps` portion so its just `test-e2e-chrome-flask` etc. 

## **Related issues**


## **Manual testing steps**
1. Compare circle CI runs to develop.
2. Verify that snaps tests run in main and flask builds

## **Screenshots/Recordings**



### **Before**
<img width="699" alt="Screenshot 2023-10-26 at 12 32 20 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/9b75e30b-ab51-4b6f-8c07-822c7603e2f4">



### **After**
<img width="688" alt="Screenshot 2023-10-26 at 12 31 37 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/69aa3edf-b3aa-4e2d-be39-33ce235075ec">




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
